docs/aws/deploy-s3.md

```diff
@@ -0,0 +1,52 @@
+# Setup an Application using Simple Storage Service (S3)
+This document explains how to setup an application that does not need a server to be running i.e. is a static application, for e.g. static websites.
+
+## Services Used
+- S3
+- CloudFront
+- Route 53
+
+## Steps
+
+### Setting up the S3 Bucket
+1. __Role__: Switch to the appropriate role in which the content for the application needs to exist in.
+2. __Create Bucket__: Click on the "Create Bucket" button in S3, give your bucket a name and select "US Standard" as the region.
+3. __Upload Content__: Click on "Upload" and drag and drop your files into the bucket.
+4. __Setup Hosting__: Next you need to setup hosting for the application. Here's how you do it:
+  1. Select the "Properties" tab and expand the "Static Website Hosting" section.
+  2. Select "Enable website hosting" and type in your application entry-point (Index Document) and the default error page you would like to show (set it to the same value of the Index Document if you don't have a custom error page).
+  3. Click "Save". You should now see an "Endpoint" url that you can access your bucket's contents at. However, this url won't be accessible until you setup permissions for the bucket as described in the next step.
+5. __Permissions__: Adequate permissions need to be set on the bucket to enable access to it. The steps for this are:
+  1. Expand the "Permissions" section in the "Properties" tab.
+  2. Select "Add Bucket Policy" and click on the "AWS Policy Generator" link in the bottom-left corner. This will open a new window/tab with the Policy Generator page.
+  3. In the Policy Generator page:
+    1. Select the "Policy Type" as "S3 Bucket Policy". 
+    2. Set the "Principal" to "*"(without quotes) to allow access to all the content in the bucket. 
+    3. For the "Action", you will need to at least select the "GetObject" permission. If you want to allow uploads to the S3 bucket from anywhere outside AWS, you should also enable the "PutObject" permission.
+    4. For the "ARN" set it to your bucket ARN which will look something like `arn:aws:s3:::<bucket_name>`
+    5. Click on "Add Statement" and then "Generate Policy". A JSON document is created for you. Copy the text from there.
+  4. Paste the JSON permission object into the text area back in the "Add Bucket Policy" dialog in the previous page and click "Save".
+  5. Verify that you can now access the bucket's content by using the url seen in the "Static Website Hosting" section.
+6. Your S3 bucket should now be setup correctly.
+
+### Setting up a CloudFront distribution
+Once you have your S3 bucket set up, you need to add a CloudFront CDN to serve the bucket's content effectively. The steps for doing so are as follows:
+
+1. __Role__: Before proceeding, make sure you switch to the appropriate role so that the role used for CloudFront is the same as the role in which the SSL certificate you would like to use exists.
```

Comment:
lets try to reword this to use the word role less.

"Before proceeding, make sure you switch to the appropriate role so that the CloudFront distribution you create is in the same account where the SSL certificate you'd like to use is stored"

Or something like that